indolaw-parser/parser_utils.py

```
@@ -318,56 +318,156 @@ def clean_law(law: List[str]) -> List[str]:
+
+        elif int_user_input_stage == 3:
+            '''
+            Stitch together plaintext lines that get separated into 2 lines due to page breaks
+            '''
+            law_stages[3][1] = clean_split_plaintext(law_stages[2][1])
+
+        elif int_user_input_stage == 4:
+            '''
+            TODO(johnamadeo): Fix "Pasal 38 B" REMOVE THE SPACE WITH REGEX
+            '''
+            law_stages[4][1] = clean_split_pasal_number(law_stages[3][1])
+
+        elif int_user_input_stage == 5:
+            '''
+            Add OPEN_QUOTE_CHAR and CLOSE_QUOTE_CHAR to PERUBAHAN_SECTION and PENJELASAN_PERUBAHAN_SECTION
+            '''
+            law_stages[5][1] = insert_perubahan_quotes(law_stages[4][1])
+
+        elif int_user_input_stage == -1:
+
+            print_section_header("Last cleaning stage finished")
+            print("Proceed to parsing?")
+            print("(y) to proceed to parsing.")
+            print("(n) to undo to specific cleaning stages.")
+            print_yes_no()
 
-    '''
-    Stitch together plaintext lines that get separated into 2 lines due to page breaks
-    '''
-    law = clean_split_plaintext(law)
+            user_input_finish = ''
 
-    save_law_to_file(law, 'e.txt')
+            while user_input_finish != 'y' and user_input_finish != 'n':
+                user_input_finish = input()
+                if user_input_finish == 'y':
+                    print(colored('Proceeding to parsing...', 'blue'))
+                    cleaning_progress['finished'] = True
+                elif user_input_finish == 'n':
+                    print(colored('Going back to cleaning stages...', 'blue'))
+                else:
+                    print(colored('Invalid input.', 'red'))
 
-    '''
-    TODO(johnamadeo): Fix "Pasal 38 B" REMOVE THE SPACE WITH REGEX
-    '''
-    law = clean_split_pasal_number(law)
+        else:
+            print(f"{colored('Invalid input.', 'red')}")
```

This is just for general error handling and we never expect ourselves to actually get here right? 

If so, `raise Exception('Some message')` instead of just printing to console.

Crashing early & often when we know we can't handle the error is the best way to eliminate silent failures

---

I see. Okay!